### ISSUE TITLE
material decomposition from mono-kev image

### ISSUE THREAD

**Issue by SpencerXia** [gjadick/multi-mat-decomp#1]
Hi, Gia.

If my understanding is right, here you get material  decomposition result from mono-kev image. [50, 65]kev. And it is an image-based material decomposition. 
1) 80kVp and 140kVp image in the last mmd function, this is confused, seems it should work for mono-kev reconstructed image.
2) Could you suggest any related paper about this decomposition algorithm for me to learn?

Thanks, appreciate that.

**Comment by gjadick** [comment]
Hi Spencer, your understanding is correct. To answer your questions:
1. That's right. I need to update this. A proper implementation should first convert the 80kVp and 140kVp images into virtual monoenergetic images and then apply the mmd function. This code was more of an exploratory implementation to start working with the mmd algorithm.
2. My code is based on the algorithm described in [this paper](https://ieeexplore.ieee.org/abstract/document/6600785) from Mendonca et al. That should have the info you need.

**Comment by SpencerXia** [comment]
> Hi Spencer, your understanding is correct. To answer your questions:
> 
> 1. That's right. I need to update this. A proper implementation should first convert the 80kVp and 140kVp images into virtual monoenergetic images and then apply the mmd function. This code was more of an exploratory implementation to start working with the mmd algorithm.
> 2. My code is based on the algorithm described in [this paper](https://ieeexplore.ieee.org/abstract/document/6600785) from Mendonca et al. That should have the info you need.

Thanks, Gia.

**Comment by SpencerXia** [comment]
My understanding is that monoenergetic images are come out after material decomposition.  I am not sure if there is better way to get mono image first

**Comment by gjadick** [comment]
Yes, I think that's right. Conventional material decomposition is into two or three basis materials. The monoenergetic images can be expressed as a linear combination of the basis material images. So it seems the multi-material (more than three materials) decomposition requires an initial two-material decomposition to get the VMIs. I do not know if there are other ways to get the virtual mono images, but that could be an interesting avenue to explore.
